plugin/main/src/kotlinx/benchmark/gradle/NativeMultiplatformTasks.kt

```diff
@@ -123,11 +125,11 @@ fun Project.createNativeBenchmarkExecTask(
 
         val binary =
             benchmarkCompilation.target.binaries.getExecutable(benchmarkCompilation.name, target.buildType)
-        val linkTask = binary.linkTask
+        val linkTask = binary.linkTaskProvider
 
         dependsOn(linkTask)
 
-        val executableFile = linkTask.outputFile.get()
+        val executableFile = linkTask.get().outputFile.get()
```

Comment:
`.get()` will eagerly configure the task and the value. 

It'd probably work instead like this:

```kotlin
onlyIf { linkTask.get().outputFile.get().exists() }
```

Reply:
Could you please help me with the other use of `executableFile` below?

Should it be `linkTask.configure { this.executable.set(it.outputFile) }` ?
This would also require changing `NativeBenchmarkExec.executable` type to `Property<File>`.

Reply:
Ah I see... yes, `NativeBenchmarkExec.executable` should be updated to be compatible with the Provider API. Do you want do that now, or in another PR?

(BTW `RegularFileProperty` or `DirectoryProperty` is preferred over `Provider<File>`.)

Reply:
I will do it in a separate PR.